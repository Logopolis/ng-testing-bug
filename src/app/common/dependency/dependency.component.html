<p *ngIf="dependencyOb$ | async as dependencyOb; else nodependency">
  {{ dependencyOb }}
</p>
<ng-template #nodependency>
  <p>The dependency observable had no value</p>
</ng-template>
