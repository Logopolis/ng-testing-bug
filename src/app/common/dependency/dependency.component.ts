import { Component } from '@angular/core';
import { MyServiceService } from '../my-service.service';

@Component({
  selector: 'app-dependency',
  templateUrl: './dependency.component.html',
  styleUrls: ['./dependency.component.scss'],
})
export class DependencyComponent {
  dependencyOb$ = this.myServiceService.myServiceData$;
  constructor(private myServiceService: MyServiceService) {}
}
