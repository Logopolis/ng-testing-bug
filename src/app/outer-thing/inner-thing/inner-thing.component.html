<p>inner-thing works!</p>
<app-dependency></app-dependency>
