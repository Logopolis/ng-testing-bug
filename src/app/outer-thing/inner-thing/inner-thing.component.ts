import { Component } from '@angular/core';

@Component({
  selector: 'app-inner-thing',
  templateUrl: './inner-thing.component.html',
  styleUrls: ['./inner-thing.component.scss'],
})
export class InnerThingComponent {}
