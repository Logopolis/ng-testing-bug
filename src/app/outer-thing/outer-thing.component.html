<p>outer-thing works!</p>
<app-inner-thing></app-inner-thing>
