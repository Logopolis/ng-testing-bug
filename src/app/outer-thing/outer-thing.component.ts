import { Component } from '@angular/core';

@Component({
  selector: 'app-outer-thing',
  templateUrl: './outer-thing.component.html',
  styleUrls: ['./outer-thing.component.scss'],
})
export class OuterThingComponent {}
