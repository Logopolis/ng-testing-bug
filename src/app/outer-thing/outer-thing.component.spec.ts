import { ComponentFixture, TestBed } from '@angular/core/testing';
import { OuterThingComponent } from './outer-thing.component';
import { InnerThingComponent } from './inner-thing/inner-thing.component';
import { DependencyComponent } from '../common/dependency/dependency.component';

import { MyServiceService } from '../common/my-service.service';
import { BehaviorSubject } from 'rxjs';

describe('OuterThingComponent', () => {
  let component: OuterThingComponent;
  let fixture: ComponentFixture<OuterThingComponent>;

  const mockMyServiceService = {
    myServiceData$: new BehaviorSubject('Test value'),
  };
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        OuterThingComponent,
        InnerThingComponent,
        DependencyComponent,
      ],
      providers: [
        { provide: MyServiceService, useValue: mockMyServiceService },
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(OuterThingComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
