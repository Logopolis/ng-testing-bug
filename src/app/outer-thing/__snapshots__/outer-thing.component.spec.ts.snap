// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuterThingComponent should match snapshot 1`] = `
<app-outer-thing>
  <p>
    outer-thing works!
  </p><app-inner-thing>
    <p>
      inner-thing works!
    </p>
    <app-dependency>
      <p>
         Test value
      </p>
    </app-dependency>
  </app-inner-thing>
</app-outer-thing>
`;
